test/parallel/test-https-argument-of-creating.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const https = require('https');
+const tls = require('tls');
+
+const dftProtocol = {};
+tls.convertNPNProtocols([ 'http/1.1' ], dftProtocol);
+
+// test for immutable `opts`
+const opts = { foo: 'bar', NPNProtocols: [ 'http/1.1' ] };
+const server1 = https.createServer(opts);
```

Comment:
Tiny nit: instead of using `server1`, `server2`, and `server3` we can use block scope.

```js
{
  // test for immutable `opts`
}

{
  // validate that `createServer` can work with the only argument requestListener
}
```